.github/workflows/test.yml

```diff
@@ -0,0 +1,24 @@
+name: Run linter
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    name: "Lint and build"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: make setup
```

Comment:
This needs better caching, or else every time you push another commit to your PR, it'll rebuild protobuf (and the SDK) from source. Looking into how to do that now. In the meantime, we can either merge this with a TODO to fix this soon, or else figure out how to fix it now.